xsnap/sources/xsnap-worker.c

```diff
@@ -205,8 +205,18 @@ static char *renderTimestamps() {
 		// been printed if the size were unlimited, not including the
 		// final \0". It writes at most size-1 characters, then writes
 		// the trailing \0.
+		// The type of tv_usec is 32 bits on Darwin and 64 bits on Linux at
+		// time of writing.
+		// We cast into the wider precision to ensure the format specifier gets
+		// the expected number of bits behind the variadic args reference.
+		// We do the same for tv_usec out of an outrageous abundance of
+		// caution.
 		wrote = snprintf(p, size, "%lu.%06lu",
-						 timestamps[i].tv_sec, timestamps[i].tv_usec);
+						 (unsigned long)timestamps[i].tv_sec,
+						 (unsigned long)timestamps[i].tv_usec);
+		if (wrote > size) {
+			return NULL;
+		}
```

Comment:
Technically redundant now we have size as signed, but it likely doesn't cost much too keep this in.

Reply:
Yeah, it’s almost sufficiently paranoid.